src/appleseed.studio/mainwindow/rendering/materialdrophandler.cpp

```diff
@@ -43,24 +43,29 @@ namespace appleseed {
 namespace studio {
 
 MaterialDropHandler::MaterialDropHandler(
-    const renderer::Project&    project,
-    RenderingManager&           rendering_manager)
-  : m_project(project), m_rendering_manager(rendering_manager)
+    renderer::Project&              project,
+    RenderingManager&               rendering_manager,
+    const renderer::ScenePicker&    scene_picker)
+  : m_project(project), m_rendering_manager(rendering_manager), m_enabled(false), m_scene_picker(scene_picker)
 {
 }
 
+void MaterialDropHandler::set_enabled(const bool enabled)
+{
+    m_enabled = enabled;
+
+    m_project.update_trace_context();
```

Comment:
This will always work because this will only happen once the `AssemblyTree` is built.
`m_project.update_trace_context();` will synchronize the `TraceContext` so that `Intersector` can use it in `ScenePicker::pick`.

The warning which I removed checked if there exists a `TraceContext` at all.
Since we are constructing a `ScenePicker` with the `RenderTab`, a `TraceContext` will come into existence with the constructor of `ScenePicker` even if the `AssemblyTree` is _not_ yet built.